x-pack/plugin/esql/src/main/java/org/elasticsearch/xpack/esql/io/stream/PlanNamedTypes.java

```diff
@@ -797,24 +797,47 @@ static EsRelation readEsRelation(PlanStreamInput in) throws IOException {
         Source source = in.readSource();
         EsIndex esIndex = readEsIndex(in);
         List<Attribute> attributes = readAttributes(in);
-        EsSourceOptions esSourceOptions = in.getTransportVersion().onOrAfter(TransportVersions.ESQL_ES_SOURCE_OPTIONS)
-            ? new EsSourceOptions(in)
-            : EsSourceOptions.NO_OPTIONS;
+        if (supportingEsSourceOptions(in.getTransportVersion())) {
+            readEsSourceOptions(in); // consume optional strings sent by remote
+        }
         boolean frozen = in.readBoolean();
-        return new EsRelation(source, esIndex, attributes, esSourceOptions, frozen);
+        return new EsRelation(source, esIndex, attributes, frozen);
     }
 
     static void writeEsRelation(PlanStreamOutput out, EsRelation relation) throws IOException {
         assert relation.children().size() == 0;
         out.writeNoSource();
         writeEsIndex(out, relation.index());
         writeAttributes(out, relation.output());
-        if (out.getTransportVersion().onOrAfter(TransportVersions.ESQL_ES_SOURCE_OPTIONS)) {
-            relation.esSourceOptions().writeEsSourceOptions(out);
+        if (supportingEsSourceOptions(out.getTransportVersion())) {
+            writeEsSourceOptions(out); // write (null) string fillers expected by remote
         }
         out.writeBoolean(relation.frozen());
     }
 
+    private static boolean supportingEsSourceOptions(TransportVersion version) {
+        return version.onOrAfter(TransportVersions.ESQL_ES_SOURCE_OPTIONS)
+            && version.before(TransportVersions.ESQL_REMOVE_ES_SOURCE_OPTIONS_8_14);
```

Comment:
BWC tests between 8.15 and 8.14 will fail with the new patch version. The new patch version 8_636_00_2 doesn't expect esSource, but 8.15 will send esSource for this patch version.

Reply:
Thanks Nhat, went with removing the version upper bound check, as you suggested. 🙏